### PR TITLE
feat: repsect oid batch arguments

### PR DESCRIPTION
**Pull request by aymanbagabas**
Preserve oid batch arguments in requset/response per [specs](https://github.com/git-lfs/git-lfs/blob/main/docs/proposals/ssh_adapter.md#requests-to-transfer-objects)

### CODE REVIEW

File: transfer/args.go [hunk 1]
```diff
@@ -19,12 +19,12 @@ const (
...
-func ParseArgs(lines []string) (map[string]string, error) {
-	args := make(map[string]string, 0)
-	for _, line := range lines {
+func ParseArgs(parts []string) (Args, error) {
+	args := make(Args, 0)
+	for _, line := range parts {
 		parts := strings.SplitN(line, "=", 2)
 		if len(parts) != 2 {
-			return nil, fmt.Errorf("invalid argument: %q", line)
+			continue
```
Comment (KyleFromKitware) [inline transfer/args.go:27]: Why are you now ignoring parts that are not key-value-pairs? Is there a specific use case for this?

**Comment by KyleFromKitware** [comment]
Looks good for the most part, I just have one question.

File: transfer/args.go (hunk 1 above)
Comment (KyleFromKitware) [reply transfer/args.go:27]: I believe "arguments" was referring to the `*argument` production before the `*oid-line` production, not non-key-value-pairs within the `oid-line` production. IMO, non-key-value pairs within an `oid-line` production (after the `oid` and `size` arguments) should still trigger an error.

Comment (aymanbagabas) [reply transfer/args.go:27]: I believe `*argument` and `oid-line` arguments use the same key-value pair format. Unknown arguments depend on the backend implementation. Perhaps you're right though, according to the specs `key=[value]`, an argument _must_ have a key followed by an `=` sign and an optional value. Thus, we should return an error here

**Review by KyleFromKitware — APPROVED** [review]

**Comment by KyleFromKitware** [comment]
Cc: @ashmckenzie

**Review by KyleFromKitware — APPROVED** [review]

**Pull request merged**